src/eduid_webapp/phone/verifications.py

```diff
@@ -69,5 +72,38 @@ def send_verification_code(user, phone):
     code, reference = new_verification_code(phone, user)
 
     current_app.msg_relay.phone_validator(reference, phone, code, user.language)
-    current_app.logger.info("Sent verification sms to user {!r}"
+    current_app.logger.info("Sent verification sms to user {}"
                             " with phone number {!s}.".format(user, phone))
+
+    
+def verify_phone_number(state, proofing_user):
+    """
+    :param proofing_user: ProofingUser
+    :param state: Phone proofing state
+
+    :type proofing_user: eduid_userdb.proofing.ProofingUser
+    :type state: PhoneProofingState
+
+    :return: None
+
+    """
+    number = state.verification.number
+    new_phone = PhoneNumber(number = number, application = 'eduid_phone',
+                            verified = True, primary = False)
+
+    has_primary = proofing_user.phone_numbers.primary
+    if has_primary is None:
+        new_phone.is_primary = True
+    try:
+        proofing_user.phone_numbers.add(new_phone)
+    except DuplicateElementViolation:
+        proofing_user.phone_numbers.find(number).is_verified = True
+        if has_primary is None:
+            proofing_user.phone_numbers.find(number).is_primary = True
+
+    save_and_sync_user(proofing_user)
+    current_app.logger.info('Mobile {} confirmed '
+                            'for user {}'.format(number, proofing_user))
+    current_app.stats.count(name='mobile_verify_success', value=1)
```

Comment:
Log message and stats talks about mobile - should be phone now. Seems to apply to a number of other log messages too. Let's keep the stats keys for now though - needs to be renamed in the stats backend at the time of update to keep old stats.

Reply:
I've fixed all log messages